gitclone/gitclone_test.go

```diff
@@ -0,0 +1,411 @@
+package gitclone
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/bitrise-io/bitrise-init/step"
+	"github.com/bitrise-io/go-utils/command"
+	"github.com/bitrise-io/go-utils/command/git"
+)
+
+const always = 9999
+const rawCmdError = "dummy_cmd_error"
+
+func Test_checkoutState(t *testing.T) {
```

Comment:
Although I see the benefit of using table driven test for this scenario this test is more than 300 lines which makes it hard to understand. Could we group the scenarios in smaller functions?

Reply:
Moved test cases out of the function, categorized them better.